lib/iris/fileformats/_nc_load_rules/helpers.py

```diff
@@ -1345,11 +1345,24 @@ def _build_dimension_coordinate(
         units=attr_units,
         bounds=bounds_data,
         attributes=attributes,
-        coord_system=coord_system,
         circular=circular,
         climatological=climatological,
     )
 
+    assert coord.var_name is not None
```

Comment:
Is it worth commenting here that this is the "add" part of what is usually a single call to `_add_or_capture` and is related to the `_add_or_capture` call in `actions.py:action_provides_grid_mapping` (assuming I've understood this correctly).